plugins/source/azure/resources/services/postgresql/servers.go

```diff
@@ -1,430 +1,158 @@
-						Resolver:    schema.PathResolver("FirewallRuleProperties.StartIPAddress"),
-					},
-					{
-						Name:        "end_ip_address",
-						Description: "The end IP address of the server firewall rule Must be IPv4 format",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("FirewallRuleProperties.EndIPAddress"),
-					},
-					{
-						Name:        "id",
-						Description: "Fully qualified resource ID for the resource Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ID"),
-					},
-					{
-						Name:        "name",
-						Description: "The name of the resource",
-						Type:        schema.TypeString,
-					},
-					{
-						Name:        "type",
-						Description: "The type of the resource Eg \"MicrosoftCompute/virtualMachines\" or \"MicrosoftStorage/storageAccounts\"",
-						Type:        schema.TypeString,
-					},
-				},
-			},
+			configurations(), firewallRules(),
 		},
 	}
 }
 
-// ====================================================================================================================
-//
-//	Table Resolver Functions
-//
-// ====================================================================================================================
-func fetchPostgresqlServers(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+func fetchPostgreSQLServers(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
 	svc := meta.(*client.Client).Services().PostgreSQL.Servers
+
 	response, err := svc.List(ctx)
 	if err != nil {
-		return diag.WrapError(err)
+		return errors.WithStack(err)
 	}
 	if response.Value == nil {
 		return nil
 	}
 	res <- *response.Value
-	return nil
-}
-func fetchPostgresqlServerPrivateEndpointConnections(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	server := parent.Item.(postgresql.Server)
-	if server.PrivateEndpointConnections == nil {
-		return nil
-	}
-	res <- *server.PrivateEndpointConnections
-	return nil
-}
-func fetchPostgresqlServerConfigurations(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	server := parent.Item.(postgresql.Server)
-	svc := meta.(*client.Client).Services().PostgreSQL.Configuration
 
-	resourceDetails, err := client.ParseResourceID(*server.ID)
-	if err != nil {
-		return diag.WrapError(err)
-	}
-	configurations, err := svc.ListByServer(ctx, resourceDetails.ResourceGroup, *server.Name)
-	if err != nil {
-		return diag.WrapError(err)
-	}
-	if configurations.Value == nil {
-		return nil
-	}
-	res <- *configurations.Value
-	return nil
-}
-
-func fetchPostgresqlServerFirewallRules(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
```

Comment:
Extracted to `plugins/source/azure/resources/services/postgresql/firewall_rules.go`